collated/main/A0138420N.md

```diff
@@ -0,0 +1,398 @@
+            try {
+                
+                redoEditTaskField.push(editTaskField.peek());
+                
+                if (redoEditTaskField.peek().equals("task")){
+                    redoEditTaskValue.push(undoEdit.getTaskName().toString()); 
+                } else if (redoEditTaskField.peek().equals("start date")){
+                    redoEditTaskValue.push(undoEdit.getStartDate().toString());
+                } else if (redoEditTaskField.peek().equals("end date")){
+                    redoEditTaskValue.push(undoEdit.getEndDate().toString()); 
+                } else if (redoEditTaskField.peek().equals("start time")){
+                    redoEditTaskValue.push(undoEdit.getStartTime().toString()); 
+                } else if (redoEditTaskField.peek().equals("end time")){
+                    redoEditTaskValue.push(undoEdit.getEndTime().toString());
+                } else if (redoEditTaskField.peek().equals("priority")){
+                    redoEditTaskValue.push(undoEdit.getPriority().toString());
+                }
+                model.editTask(undoEdit, editTaskField.pop(), editTaskValue.pop());
+            } catch (TaskNotFoundException e) {
+                e.printStackTrace();
+            } catch (IllegalValueException ive) {
+                new CommandResult(ive.getMessage());
+            }
+        }
+        model.updateListing();
+        return new CommandResult(String.format(MESSAGE_UNDO_COMMAND_SUCCESS, previousCommand));
+    }
+    
+    @Override
+    public String toString() {
+        // TODO Auto-generated method stub
+        return null;
+    }
+
+}
+```
+###### /main/java/seedu/ggist/logic/parser/Parser.java
+``` java
+                
+        case UndoCommand.COMMAND_WORD:
+            return new UndoCommand();
+            
+        case RedoCommand.COMMAND_WORD:
+            return new RedoCommand();
+            
+```
+###### /main/java/seedu/ggist/logic/parser/Parser.java
+``` java
+    private Command prepareDelete(String args) {
+        String[] parts = args.split(",");
+        ArrayList<Integer> indexes = new ArrayList<Integer>();
+        for (int i = 0; i < parts.length; i++) {
+            try {
+                indexes.add(Integer.parseInt(parts[i].trim()));
+            } catch (NumberFormatException nfe) {
+                return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE));
+            }
+        }
+        
+        for(int i = 0; i < parts.length; i++){
+        Optional<Integer> index = parseIndex(parts[i]);
+        if(!index.isPresent()){
+            return new IncorrectCommand(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE));
+            }
+        }
+        
+        return new DeleteCommand(indexes);
+    }
+```
+###### /main/java/seedu/ggist/logic/parser/Parser.java
+``` java
+    private Command prepareDone(String args) {
+        String[] parts = args.split(",");
```

Comment:
these few lines are common to each method; why not extract them to a method by itself?